oci/repositories.bzl

```diff
@@ -37,8 +39,81 @@ crane_repositories = repository_rule(
     },
 )
 
+
+ZOT_BUILD_TMPL = """\
+# Generated by container/repositories.bzl
+load("@contrib_rules_oci//oci:toolchain.bzl", "registry_toolchain")
+registry_toolchain(
+    name = "zot_toolchain", 
+    registry = "zot",
+    launcher = "launcher.sh"
+)
+"""
+
+ZOT_LAUNCHER_TMPL = """\
+#!/usr/bin/env bash
+set -o errexit -o nounset -o pipefail
+
+SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+ZOT="${SCRIPT_DIR}/zot"
+
+function start_registry() {
+    local CONFIG_PATH="$1/config.json"
+    cat > "${CONFIG_PATH}" <<EOF
+{
+    "storage": {"rootDirectory": "$1" },
+    "http": { "port": "0", "address": "127.0.0.1" },
+    "log": { "level": "info" }
+}
+EOF
+
+    local OUTPUT="$2"
+
+    "${ZOT}" serve "${CONFIG_PATH}" 2>&1 >> "${OUTPUT}" &
+
+    local DEADLINE=5
+    local TIMEOUT=$((SECONDS+${DEADLINE}))
+
+    while [ "${SECONDS}" -lt "${TIMEOUT}" ]; do
+        PORT=$(cat $OUTPUT | sed -nr 's/.+"port":([0-9]+),.+/\\1/p')
+        if [ -n "${PORT}" ]; then
+            break
+        fi
+    done
+    if [ -n "${PORT}" ]; then
+        echo "Exhausted: registry couldn't become ready within ${DEADLINE}s." >> "${OUTPUT}"
+    fi
+    REGISTRY="127.0.0.1:${PORT}"
+}
+"""
+
+def _zot_repo_impl(repository_ctx):
+    platform = repository_ctx.attr.platform.replace("x86_64", "amd64").replace("_", "-")
```

Comment:
seems kinda gross to have this replace here, will grow more cases. It also seems like a design flaw - shouldn't we register a toolchain for *all* platforms like we do in other rulesets, and defer selection until bazel knows the target platform?

Reply:
the ideal solution is here to have `PLATFORMS` for each toolchain type as we have a sparse tool. We do defer the toolchain selection here. this comes from release artifact having different names for `amd64`. 

zot uses `amd64` for `x86_64` while crane uses `x86_64` for `amd64`.  `go-containerregistry_Darwin_x86_64.tar.gz` vs `zot-darwin-amd64`